homeassistant/components/metoffice/weather.py

```diff
@@ -1,29 +1,49 @@
 """Support for UK Met Office weather service."""
+
 import logging
 
 import datapoint as dp
 import voluptuous as vol
 
-from homeassistant.components.weather import PLATFORM_SCHEMA, WeatherEntity
+from homeassistant.components.weather import (
+    ATTR_FORECAST_CONDITION,
+    ATTR_FORECAST_PRECIPITATION,
+    ATTR_FORECAST_TEMP,
+    ATTR_FORECAST_TIME,
+    ATTR_FORECAST_WIND_BEARING,
+    ATTR_FORECAST_WIND_SPEED,
+    PLATFORM_SCHEMA,
+    WeatherEntity,
+)
 from homeassistant.const import (
     CONF_API_KEY,
     CONF_LATITUDE,
     CONF_LONGITUDE,
+    CONF_MODE,
     CONF_NAME,
     TEMP_CELSIUS,
 )
 from homeassistant.helpers import config_validation as cv
 
-from .sensor import ATTRIBUTION, CONDITION_CLASSES, MetOfficeCurrentData
+from .const import (
+    ATTRIBUTION,
+    CONDITION_CLASSES,
+    DEFAULT_NAME,
+    MODE_3HOURLY,
+    MODE_DAILY,
+    VISIBILITY_CLASSES,
+)
+from .data import MetOfficeData
 
 _LOGGER = logging.getLogger(__name__)
 
-DEFAULT_NAME = "Met Office"
-
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_API_KEY): cv.string,
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_MODE, default=MODE_3HOURLY): vol.In(
```

Comment:
We no longer accept additions or changes to the yaml config of integrations. See: https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md

If you want to change the configuration, please first make a separate PR where the integration is refactored to use config flow and config entries. When that is merged we can rebase here and continue.

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

Reply:
Right-o. Can you point me to an example of how to do this for something like the weather integrations, please, as I've been struggling to do this over the last few days? All the documentation assumes a nice simple single entry-point entity, but of course the weather integrations are both sensor and entity, so they don't work quite the same.

I'm happy to start a new branch from the current "dev", without my changes on, to try and get this done first... but I could do with a working example of how to get the config_flow working for platform schema based projects as a template.

Any thoughts?

Reply:
The nws integration has three different endpoints but just a weather platform. It's easy to add a sensor platform to that though. Just forward the entry to the sensor domain too, and then have the sensor platform entities subscribe to updates from the data update coordinator(s).

https://github.com/home-assistant/core/blob/1c84905bcd05d27d642f1e6a6a8c9ab553b45b2f/homeassistant/components/nws/__init__.py#L66-L111